### PR TITLE
Add manual deployment guide for ScalarDB Cluster

### PR DESCRIPTION
**Pull request by kota2and3kan**
This PR adds the manual deployment guide for ScalarDB Cluster on EKS.
Also, I marked ScalarDB Server and ScalarDB GraphQL Server as `Deprecated` since it is recommended to use ScalarDB Cluster instead of them.

**Note:**
* At this time, we provide ScalarDB Cluster on AWS Marketplace only. So, I don't create the document for AKS (Although, I added the images for AKS). After we will release ScalarDB Cluster on Azure Marketplace, I will add documents for AKS in the future.
* ScalarDB Cluster is not available on AWS Marketplace yet. So, we should merge this PR after ScalarDB Cluster will be released on AWS Marketplace.

Please take a look!

### CODE REVIEW

File: docs/CreateEKSClusterForScalarDB.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# Guidelines for creating an EKS cluster for ScalarDB Server
+# (Deprecated) Guidelines for creating an EKS cluster for ScalarDB Server
+
+**Note:** ScalarDB Server is now deprecated. Please use [ScalarDB Cluster](./ManualDeploymentGuideScalarDBClusterOnEKS.md) instead.
```
Comment (kota2and3kan) [inline docs/CreateEKSClusterForScalarDB.md:3]: ScalarDB Server is now deprecated. So, I added notes for that in existing documents for ScalarDB Server.

File: docs/CreateEKSClusterForScalarDBCluster.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+* Create the EKS cluster by using Kubernetes version 1.21 or higher.
+* Configure the EKS cluster based on the version of Kubernetes and your project's requirements.
+
+## Recommendations (optional)
+
+The following are some recommendations for deploying ScalarDB Cluster. These recommendations are not required, so you can choose whether or not to apply these recommendations based on your needs.
+
+### Use 4vCPU / 8GB memory nodes for the worker node in the ScalarDB Cluster node group
+
+From the perspective of commercial licenses, resources for one pod running ScalarDB Cluster are limited to 2vCPU / 4GB memory. In addition, we recommend deploying one ScalarDB Cluster pod and one Envoy pod on one worker node. Note that the Envoy pod does not exist when you use `direct-kubernetes` mode.
```
Comment (kota2and3kan) [inline docs/CreateEKSClusterForScalarDBCluster.md:22]: This part is a bit different from ScalarDB Server. ScalarDB Cluster has two client modes. If you use `direct-kubernetes` mode, you don't need to deploy Envoy.

File: docs/K8sMonitorGuide.md [hunk 3]
```diff
@@ -74,17 +74,19 @@ scalar-monitoring-kube-pro-operator-865bbb8454-9ppkc     1/1     Running   0
...
-   * [ScalarDL Auditor](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-scalardl-auditor.md#prometheusgrafana-configurations)
+   * [ScalarDB Cluster](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-scalardb-cluster.md#prometheus-and-grafana-configurations--recommended-in-production-environments)
+   * [(Deprecated) ScalarDB Server](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-scalardb.md#prometheusgrafana-configurations--recommended-in-the-production-environment)
+   * [(Deprecated) ScalarDB GraphQL](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-scalardb-graphql.md#prometheusgrafana-configurations-recommended-in-the-production-environment)
+   * [ScalarDL Ledger](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-scalardl-ledger.md#prometheusgrafana-configurations-recommended-in-the-production-environment)
+   * [ScalarDL Auditor](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-scalardl-auditor.md#prometheusgrafana-configurations-recommended-in-the-production-environment)
```
Comment (kota2and3kan) [inline docs/K8sMonitorGuide.md:81]: These changes are not related to ScalarDB Cluster. I just fixed the broken links here.

File: docs/ManualDeploymentGuideScalarDBClusterOnEKS.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Deploy ScalarDB Cluster on Amazon Elastic Kubernetes Service (EKS)
+
+This guide explains how to deploy ScalarDB Cluster on Amazon Elastic Kubernetes Service (EKS).
+
+In this guide, you will create one of the following two environments in your AWS environment. The difference between the two environments is which [client modes](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#client-modes) you use:
```
Comment (kota2and3kan) [inline docs/ManualDeploymentGuideScalarDBClusterOnEKS.md:5]: There are two client modes in ScalarDB Cluster. So, there are two patterns for deployment on the Kubernetes environment.

File: docs/ManualDeploymentGuideScalarDBClusterOnEKS.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+After deploying ScalarDB Cluster in your EKS cluster, you must check the status of each component. For details, see [Components to Regularly Check When Running in a Kubernetes Environment](./RegularCheck.md).
+
+## Step 8. Monitor your ScalarDB Cluster deployment
+
+After deploying ScalarDB Cluster in your EKS cluster, we recommend monitoring the deployed components and collecting their logs, especially in production. For details, see [Monitoring Scalar products on a Kubernetes cluster](./K8sMonitorGuide.md) and [Collecting logs from Scalar products on a Kubernetes cluster](./K8sLogCollectionGuide.md).
+
+## Step 9. Deploy your application
+
+If you use [`direct-kubernetes` client mode](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#direct-kubernetes-client-mode), you must deploy additional Kubernetes resources. For details, see [Deploy your client application on Kubernetes with `direct-kubernetes` mode](https://github.com/scalar-labs/helm-charts/blob/main/docs/how-to-deploy-scalardb-cluster.md#deploy-your-client-application-on-kubernetes-with-direct-kubernetes-mode).
```
Comment (kota2and3kan) [inline docs/ManualDeploymentGuideScalarDBClusterOnEKS.md:55]: This part is a ScalarDB Cluster specific step. Especially, if you use the `direct-kubernetes` client mode, you must deploy some resources like `Role`, `RoleBinding`, and `ServiceAccount` to make the `direct-kubernetes` mode client work properly.

There is an existing document for this on the Helm document side. So, I just added the link to that document.

File: docs/images/drawio/AKS_ScalarDB_Cluster_Direct_Kubernetes_Mode.drawio [hunk 6]
```diff
@@ -0,0 +1,320 @@
+<mxfile host="app.diagrams.net" modified="2023-07-21T07:32:41.001Z" agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36" etag="8UMnNdPi2g1rvwuZ5w2Y" version="21.6.5" type="google">
```
Comment (kota2and3kan) [inline docs/images/drawio/AKS_ScalarDB_Cluster_Direct_Kubernetes_Mode.drawio:1]: This is a diagram for AKS. We don't provide ScalarDB Cluster on the Azure Marketplace yet, but I added the diagram only for future releases.

File: docs/ManualDeploymentGuideScalarDBClusterOnEKS.md (hunk 4 above)
Comment (josh-wong) [inline docs/ManualDeploymentGuideScalarDBClusterOnEKS.md:5]:
```suggestion
In this guide, you will create one of the following two environments in your AWS environment. The environments differ depending on which [client mode](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#client-modes) you use:
```

File: docs/ManualDeploymentGuideScalarDBClusterOnEKS.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
+# Deploy ScalarDB Cluster on Amazon Elastic Kubernetes Service (EKS)
+
+This guide explains how to deploy ScalarDB Cluster on Amazon Elastic Kubernetes Service (EKS).
+
+In this guide, you will create one of the following two environments in your AWS environment. The difference between the two environments is which [client modes](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#client-modes) you use:
+
+* Use [`direct-kubernetes` client mode](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#direct-kubernetes-client-mode). In this mode, you deploy your application in the same EKS cluster as your ScalarDB Cluster deployment.
```
Comment (josh-wong) [inline docs/ManualDeploymentGuideScalarDBClusterOnEKS.md:7]:
```suggestion
* **[`direct-kubernetes` client mode](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#direct-kubernetes-client-mode).** In this mode, you deploy your application in the same EKS cluster as your ScalarDB Cluster deployment.
```

File: docs/ManualDeploymentGuideScalarDBClusterOnEKS.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+  ![image](./images/png/EKS_ScalarDB_Cluster_Direct_Kubernetes_Mode.drawio.png)  
+
+* Use [`indirect` client mode](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#indirect-client-mode). In this mode, you deploy your application in an environment that is different from the EKS cluster that contains your ScalarDB Cluster deployment.
```
Comment (josh-wong) [inline docs/ManualDeploymentGuideScalarDBClusterOnEKS.md:11]:
```suggestion
* **[`indirect` client mode](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#indirect-client-mode).** In this mode, you deploy your application in an environment that is different from the EKS cluster that contains your ScalarDB Cluster deployment.
```

File: docs/ManualDeploymentGuideScalarDBClusterOnEKS.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+  ![image](./images/png/EKS_ScalarDB_Cluster_Indirect_Mode.drawio.png)
+
+## Step 1. Subscribe to ScalarDB Cluster in AWS Marketplace
+
+You must get the ScalarDB Cluster container image by visiting [AWS Marketplace](https://aws.amazon.com/marketplace/pp/prodview-rzbuhxgvqf4d2) and subscribing to ScalarDB Cluster. For details on how to subscribe to ScalarDB Cluster in AWS Marketplace, see [Subscribe to Scalar products from AWS Marketplace](./AwsMarketplaceGuide.md#subscribe-to-scalar-products-from-aws-marketplace).
```
Comment (josh-wong) [inline docs/ManualDeploymentGuideScalarDBClusterOnEKS.md:17]: When I go to the [AWS Marketplace](https://aws.amazon.com/marketplace/pp/prodview-rzbuhxgvqf4d2) link, I see a page that shows that the latest version is `ScalarDB Server 3.9`. I worry this might be confusing for users if they are expecting to see `ScalarDB Cluster`.

**Review by josh-wong — CHANGES_REQUESTED** [review]
I've added some comments and suggestions. PTAL!

File: docs/ManualDeploymentGuideScalarDBClusterOnEKS.md (hunk 9 above)
Comment (kota2and3kan) [reply docs/ManualDeploymentGuideScalarDBClusterOnEKS.md]: Ah, sorry. I overlooked this part.
We have to update the URL since the `ScalarDB (BYOL)` and `ScalarDB Cluster [Standard|Premium] Edition` are different products in AWS Marketplace.
I updated the URL in c10051a9403bf875578bad7402f581318b49d1d1.

**Comment by kota2and3kan** [comment]
@josh-wong 
Thank you for your review!
I applied your suggestions and left one comment.
Please take a look again when you have time!

**Review by josh-wong — APPROVED** [review]
LGTM! Thank you!🙇‍♂️

File: docs/ManualDeploymentGuideScalarDBClusterOnEKS.md (hunk 9 above)
Comment (josh-wong) [reply docs/ManualDeploymentGuideScalarDBClusterOnEKS.md]: Thanks! I can't confirm the contents of the AWS Marketplace page when I'm not logged in, but I think that might be intentional.

Comment (josh-wong) [reply docs/ManualDeploymentGuideScalarDBClusterOnEKS.md]: Got it. Thanks!

File: docs/AwsMarketplaceGuide.md [hunk 10]
```diff
@@ -1,12 +1,14 @@
 # How to install Scalar products through AWS Marketplace
 
-Scalar products (ScalarDB, ScalarDL, and their tools) are available in the AWS Marketplace as container images. This guide explains how to install Scalar products through the AWS Marketplace.
+Scalar products (ScalarDB Server, ScalarDB Cluster, ScalarDL, and their tools) are available in the AWS Marketplace as container images. This guide explains how to install Scalar products through the AWS Marketplace.
```
Comment (feeblefakie) [inline docs/AwsMarketplaceGuide.md:3]:
```suggestion
Scalar products (ScalarDB, ScalarDL, and their tools) are available in the AWS Marketplace as container images. This guide explains how to install Scalar products through the AWS Marketplace.
```
For here, I think it's probably better and consistent to just say ScalarDB and ScalarDL.

File: docs/AwsMarketplaceGuide.md [hunk 11]
```diff
@@ -55,6 +57,27 @@ By subscribing to Scalar products in the AWS Marketplace, you can pull the conta
 
    You need to specify the private container registry (ECR) of the AWS Marketplace and the version (tag) as the value for `[].image.repository` and `[].image.version (tag)` in the custom values file. You also need to specify the service account name that you created in the previous step as the value for `[].serviceAccount.serviceAccountName` and set `[].serviceAccount.automountServiceAccountToken` to `true`.
 
+   * ScalarDB Cluster Examples
```
Comment (feeblefakie) [inline docs/AwsMarketplaceGuide.md:60]: @josh-wong 
I think it's something we should consider applying tabs.
We can do it in another PR. Anyway, let's discuss it.

File: docs/AwsMarketplaceGuide.md [hunk 12]
```diff
@@ -1,12 +1,14 @@
...
 Note that some Scalar products are available under commercial licenses, and the AWS Marketplace provides those products as pay-as-you-go pricing or a Bring Your Own License (BYOL) option. If you select pay-as-you-go pricing, AWS will charge you our product license fee based on your usage. If you select BYOL, please make sure you have the appropriate license for the product.
 
 ## Subscribe to Scalar products from AWS Marketplace
 
 1. Access to the AWS Marketplace.
+   * [ScalarDB Cluster Standard Edition (Pay-As-You-Go)](https://aws.amazon.com/marketplace/pp/prodview-jx6qxatkxuwm4)
+   * [ScalarDB Cluster Premium Edition (Pay-As-You-Go)](https://aws.amazon.com/marketplace/pp/prodview-djqw3zk6dwyk6)
    * [ScalarDL Ledger (Pay-As-You-Go)](https://aws.amazon.com/marketplace/pp/prodview-wttioaezp5j6e)
    * [ScalarDL Auditor (Pay-As-You-Go)](https://aws.amazon.com/marketplace/pp/prodview-ke3yiw4mhriuu)
    * [ScalarDB (BYOL)](https://aws.amazon.com/marketplace/pp/prodview-rzbuhxgvqf4d2)
```
Comment (feeblefakie) [inline docs/AwsMarketplaceGuide.md:14]: For ScalarDB (BYOL), it seems to be the correct name, but maybe we should call it `ScalarDB Server (BYOL)` to make it matchable with other docs?
Also, shouldn't it be marked as deprecated?
Or, is it going to be replaced with `ScalarDB Cluster (BYOL)`?

**Review by feeblefakie — COMMENTED** [review]
Overall, looking good. Thank you!
Left some comments. PTAL!

File: docs/AwsMarketplaceGuide.md (hunk 11 above)
Comment (kota2and3kan) [reply docs/AwsMarketplaceGuide.md:60]: I agree. In other documents, there also are several similar parts that describe configuration examples in each product. I think it is better to use `tabs` in such examples part.

File: docs/AwsMarketplaceGuide.md (hunk 12 above)
Comment (kota2and3kan) [reply docs/AwsMarketplaceGuide.md:14]: @feeblefakie  
Thank you for your review!


> For ScalarDB (BYOL), it seems to be the correct name, but maybe we should call it ScalarDB Server (BYOL) to make it matchable with other docs?

On the AWS Marketplace page, the product name (page title) is `ScalarDB (BYOL)` at this time. So, I use the same name in the document.

Also, when we published this product, there was no other product like ScalarDB Cluster. There was ScalarDB Server only. So, I didn't state `ScalarDB Server` explicitly before.

I agree that we should explain `ScalarDB Server` explicitly now. But, first, we have to update the product name on AWS Marketplace.

So, I will do it first, and after that, I will update the product name from `ScalarDB (BYOL)` to `ScalarDB Server (BYOL)` in another PR.


> Also, shouldn't it be marked as deprecated?
> Or, is it going to be replaced with ScalarDB Cluster (BYOL)?

I think we should mark it as deprecated. (And, in the future, we can remove it from document.)
However, at this time, there is one minor problem.

Now, ScalarDB Cluster (Pay As You Go) is provided in AWS Marketplace only. In Azure Marketplace, ScalarDB Cluster is not published.

So, from the perspective of Azure Marketplace users, they cannot use ScalarDB Cluster. So, I think we should not mark ScalarDB Server as deprecated in Azure Marketplace Guide yet.

Also, I thought we can mark ScalarDB Server as deprecated in the AWS Marketplace Guide only. But, I also thought it causes some confusion since AWS Marketplace Guide states `it is deprecated` and Azure Marketplace Guide states `it is not deprecated`.

This is why I didn't mark ScalarDB Server as deprecated in the AWS Marketplace Guide.

However, I already explain `ScalarDB Server is deprecated` in several documents. So, now I'm thinking about it is OK to mark ScalarDB Server as deprecated in AWS Marketplace Guide. 

Also, I think we can add `[Deprecated]` mark in the AWS Marketplace page.

As the above, there is a bit complex concerns between the each Marketplace. I think we also must to consider how to provide ScalarDB Cluster for Azure users. So, could you please let me talk with you about these points later?

Comment (kota2and3kan) [reply docs/AwsMarketplaceGuide.md]: @feeblefakie  
According to the discussion, I updated the product name on the AWS Marketplace side to `[Deprecated] ScalarDB Server (BYOL)`.
https://aws.amazon.com/marketplace/pp/prodview-rzbuhxgvqf4d2

Same as the above update, I updated this document in commit 3fb947ebd24be61fbe58b157e2abb19fdd422a1f.

And, we will keep this `[Deprecated] ScalarDB Server (BYOL)` for a while. (Remove it in the future after ScalarDB Server is completely deprecated.)

Also, after we implement BYOL license key feature, we will add a new product `ScalarDB Cluster (BYOL)` in the future.

File: docs/AwsMarketplaceGuide.md (hunk 11 above)
Comment (josh-wong) [reply docs/AwsMarketplaceGuide.md:60]: Yes, we can definitely add tabs to make parts like this easier to follow. I will consider how to implement tabs in this case (and in similar docs, like for Helm Charts).

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by feeblefakie** [comment]
@kota2and3kan Can you fix the conflict? Once it's resolved, you can merge it.
